From d981e111c5213728cab5706008964093fe668373 Mon Sep 17 00:00:00 2001
From: Huynh Thanh Hung <hung.huynh.wz@renesas.com>
Date: Mon, 3 Aug 2020 15:05:59 +0700
Subject: [PATCH] arm64: dts: r8a774e1-hihope-rzg2h: Re-order CMA area and
 remove lossy area.

We will disable ECC for CMA area, we should move CMA area to start
address of block size 256MB to not using extra split mode.
Lossy feature can't be used when enable ECC, so that we can remove it.

Signed-off-by: Huynh Thanh Hung <hung.huynh.wz@renesas.com>
---
 arch/arm64/boot/dts/renesas/r8a774e1-hihope-rzg2h.dts | 14 ++++----------
 1 file changed, 4 insertions(+), 10 deletions(-)

diff --git a/arch/arm64/boot/dts/renesas/r8a774e1-hihope-rzg2h.dts b/arch/arm64/boot/dts/renesas/r8a774e1-hihope-rzg2h.dts
index 0566bfa..185550c 100644
--- a/arch/arm64/boot/dts/renesas/r8a774e1-hihope-rzg2h.dts
+++ b/arch/arm64/boot/dts/renesas/r8a774e1-hihope-rzg2h.dts
@@ -29,17 +29,11 @@
 		#size-cells = <2>;
 		ranges;
 
-		/* device specific region for Lossy Decompression */
-		lossy_decompress: linux,lossy_decompress@54000000 {
-			no-map;
-			reg = <0x00000000 0x54000000 0x0 0x03000000>;
-		};
-
 		/* global autoconfigured region for contiguous allocations */
-		global_cma: linux,cma@58000000 {
+		global_cma: linux,cma@50000000 {
 			compatible = "shared-dma-pool";
 			reusable;
-			reg = <0x00000000 0x58000000 0x0 0x20000000>;
+			reg = <0x00000000 0x50000000 0x0 0x20000000>;
 			linux,cma-default;
 		};
 
@@ -47,13 +41,13 @@
 		mmp_reserved: linux,multimedia@70000000 {
 			compatible = "shared-dma-pool";
 			reusable;
-			reg = <0x00000000 0x78000000 0x0 0x10000000>;
+			reg = <0x00000000 0x70000000 0x0 0x10000000>;
 		};
 	};
 
 	mmngr {
 		compatible = "renesas,mmngr";
-		memory-region = <&mmp_reserved>, <&lossy_decompress>;
+		memory-region = <&mmp_reserved>;
 	};
 
 	mmngrbuf {
-- 
1.9.1

